src/states/Runes.ts

```diff
@@ -310,7 +310,11 @@ export class OvaleRunesClass extends States<RuneData> implements StateModule {
         this.profiler.StartProfiling("OvaleRunes_state_GetRunesCooldown");
         for (let slot = 1; slot <= RUNE_SLOTS; slot += 1) {
             const rune = state.rune[slot];
-            usedRune[slot] = rune.endCooldown - atTime;
+            if (IsActiveRune(rune, atTime)) {
```

Comment:
I'm not sure what is the point of this change. Could you explain? A negative cooldown should not be a problem and may be a good thing if mathematical operations are done on the result.

Btw, the TimeToRunes condition code is dubious. Do you know why it returns a constant?

Reply:
The time that `GetRunesCooldown()` was returning wasn't accurate, even for a negative cooldown because `rune.endCooldown` is set to 0 when the rune is active/available, so it was returning `0 - atTime`, which didn't make sense.  The conditions that forward to `GetRunesCooldown()` did their own clamping to ensure the time was greater than zero, but this change was just for correctness.

Reply:
The `TimeToRunes()` condition looks correct to me. It's meant to return the number of seconds until `N` runes are available, which is exactly what `GetRunesCooldown()` returns.

Reply:
> The time that `GetRunesCooldown()` was returning wasn't accurate, even for a negative cooldown because `rune.endCooldown` is set to 0 when the rune is active/available, so it was returning `0 - atTime`, which didn't make sense. The conditions that forward to `GetRunesCooldown()` did their own clamping to ensure the time was greater than zero, but this change was just for correctness.
Maybe it means that endCooldown should not be set to 0 when a rune is inactive. The idea is that scripts may want to perform checks like "2 seconds after the cooldown ended" which is not possible this way.

Reply:
> The `TimeToRunes()` condition looks correct to me. It's meant to return the number of seconds until `N` runes are available, which is exactly what `GetRunesCooldown()` returns.

I mean, it should be a a value that decrease at the rate of -1 per second

Reply:
Oh! Yes, you are right. We should fix that. I can submit a pullup in a separate commit.

Reply:
> > The time that `GetRunesCooldown()` was returning wasn't accurate, even for a negative cooldown because `rune.endCooldown` is set to 0 when the rune is active/available, so it was returning `0 - atTime`, which didn't make sense. The conditions that forward to `GetRunesCooldown()` did their own clamping to ensure the time was greater than zero, but this change was just for correctness.
> 
> Maybe it means that endCooldown should not be set to 0 when a rune is inactive. The idea is that scripts may want to perform checks like "2 seconds after the cooldown ended" which is not possible this way.

Yes, that's definitely an enhancement we can do, but it's not how that Runes module is currently designed. I think changing it wouldn't be too hard, though.